src/components/organisms/ProfileModal/EditProfileLinkForm/EditProfileLinkForm.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useCallback, useMemo } from "react";
+import { useForm } from "react-hook-form";
+import * as Yup from "yup";
+
+import { updateProfileLinks } from "api/profile";
+
+import { ProfileLink, User } from "types/User";
+
+import { WithId } from "utils/id";
+import { isDefined } from "utils/types";
+
+import Button from "components/atoms/Button";
+import { InputField } from "components/atoms/InputField";
+
+import { UserProfileMode } from "../ProfilePopoverContent";
+
+import "./EditProfileLinkForm.scss";
+
+export interface EditProfileLinkFormProps {
+  edit?: boolean;
+  user?: WithId<User>;
+  profileLink?: ProfileLink;
+  setUserProfileMode: (mode: UserProfileMode) => void;
+}
+
+export const EditProfileLinkForm: React.FC<EditProfileLinkFormProps> = ({
+  edit = false,
+  user,
+  profileLink,
+  setUserProfileMode,
+}) => {
+  const title = edit ? "Edit your profile link" : "Add a profile link";
+  const saveButtonText = edit ? "Save changes" : "Add link";
+
+  const validateUniqueTitle = useCallback(
+    (title) => {
+      const allowedCountOfSameTitles = edit ? 1 : 0;
+      const sameTitleCount =
+        user?.profileLinks?.filter(
+          (userProfileLink) =>
+            userProfileLink.title === profileLink?.title ||
+            userProfileLink.title === title
+        ).length ?? 0;
+
+      return sameTitleCount === allowedCountOfSameTitles;
+    },
+    [profileLink, user?.profileLinks, edit]
+  );
```

Comment:
It looks too complicated, but it solves the edgecase, if a person renames the Profile link title into an existing one. I can't suggest a better approach now, but I'm sure there is one @0xdevalias @sunny-viktoryia

Reply:
Agreed that this looks way too convoluted/complicated. You're also shadowing variables like `title`/etc here which makes the code way harder to reason about.

It's hard to even follow the logic of what is intended here to then be able to re-write it cleaner.

Reply:
This is used in the `Yup` `validationSchema` defined inline below, as the handler to `Yup`'s `.test`:

- https://github.com/jquense/yup#mixedtestname-string-message-string--function-test-function-schema

Basically the handler is passed the `value` to be tested + an optional `context` that contains `path`, `schema`, `options`, `parent`, `originalValue`, `createError`

---

`user` is of type `WithId<User>`, optionally passed in from props, so may be `undefined`

---

`validationSchema` is passed into `useForm`.. which i'm not even sure is correct based on what I am seeing in the usage docs?

- https://react-hook-form.com/api/useform

Reply:
Looking at `node_modules/react-hook-form/dist/useForm.d.ts:4` it definitely mentions `validationSchema` there + we use it across the codebase in as similar way.. maybe the API changed between the version we are using + the latest docs on the website?

Reply:
Ok, so I think the intent of this is that:
- if we're in `edit` mode, then there can be 1 matching duplicate title (presumably intended to be the current title being edited)
- if we're not in edit mode, then there can't be any duplicates

We get passed the `user` and the `profileLink`.

`ProfileLink` has a `title` and `url`.

Reply:
I'm not really sure of the mechanics/usage of `edit` and didn't want to dig further trying to get the context in my head, but something like this should work and be far simpler:

```typescript
const validateUniqueTitle = useCallback(
  (checkTitle: string) => {
    if (title === profileLink.title) return true

    const isDuplicateTitle = user?.profileLinks?.find(link => link.title === title) ?? false

    return !isDuplicateTitle
  [profileLink, user?.profileLinks]
);
```

Reply:
@0xdevalias OMG!!!! SO EASY!!! THANK YOU! ❤️ 
I was also not happy with the way I implemented that... 

Amazing fix here: https://github.com/sparkletown/sparkle/pull/1499/commits/4b9c4fee3c3c8ac3d75c4a18814995447bfa29da#diff-7e60858c69455cc177571d9c3680b6df13c67916f9827da273b5e0bf46ee9ee7R36-R42